include/counter.h

```diff
@@ -27,30 +27,71 @@
 extern "C" {
 #endif
 
-typedef void (*counter_callback_t)(struct device *dev, void *user_data);
+/** @brief Single shot callback
+ *
+ * @param dev       Pointer to the device structure for the driver instance.
+ * @param id        Single shot request id.
+ * @param counter   Counter value that triggered the callback.
+ * @param user_data User data.
+ */
+typedef void (*counter_single_shot_callback_t)(struct device *dev,
+					       u32_t id,
+					       u32_t counter,
+					       void *user_data);
+
+struct counter_single_shot {
+	/** Index of the request. Maximal index is implementation dependent.*/
+	u8_t id;
+	/** If true ticks are treated as absolute value, else it is relative to
+	 * the counter reading performed during the call.
```

Comment:
I would assume relative should be the only way. I don't get why there would be 2 different ways.

Reply:
Absolute value is needed to implement precise periodic timer as you have to schedule new event in relation to last event compare value, not current tick count.